api/src/modules/assignor/assignor.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/database/prisma.service';
+import { AssignorDTO } from '../../dtos/assignor.dto';
+
+
+@Injectable()
+export class AssignorService {
+  constructor(private prisma: PrismaService) {}
+
+  async create(data: AssignorDTO) {
+    const assignorExists = await this.prisma.assignor.findFirst({
+      where: {
+        document: data.document,
+      },
+    });
+
+    if (assignorExists) {
+      throw new Error('Assignor already exists');
+    }
+
+    const assignor = await this.prisma.assignor.create({
+      data,
+    });
+ 
+    return assignor;
+  }
+
+  async findAll() {
+   const assignors = await this.prisma.assignor.findMany({
+      include: {
+        receivables: true,
+      },
+    });
+    return assignors;
+  }
+
+  async update(id: string, data: AssignorDTO) {
+    const assignotExists = await this.prisma.assignor.findUnique({
+      where: {
+        id,
+      },
+    });
+
+    if (!assignotExists) {
+      throw new Error('Assignor not found');
+    }
+
+    return await this.prisma.assignor.update({
```

Comment:
o "return await" é redundante aqui, poderia passar diretamente o return this.prisma....